storage/range_command.go

```diff
@@ -681,6 +681,8 @@ func (r *Range) InternalLeaderLease(batch engine.Engine, ms *proto.MVCCStats, ar
 	// extra tick. This allows multiple requests from the same replica to
 	// merge without ticking away from the minimal common start timestamp.
 	if prevLease.RaftNodeID == 0 || isExtension {
+		// TODO(tschottdorf) probably go all the way back to
+		// prevLease.Start() (so it's properly extending previous lease).
```

Comment:
no, the relevant case is the one in which this is the own lease (`isExtension==true`). I've updated the comment.